### DISCUSSION TITLE
jsMind  vs. GoJS Mind Map

### DISCUSSION THREAD

**Discussion by ipader** [hizzgdev/jsmind#564] [General]
首先，很感谢作者开源如此简洁功能又完整的思维导图项目。本人观察到除了本项目外，还有一款基于 GoJs 的 Mind Map 组件，它的优势是在视图渲染上似乎更胜一筹，比如 拖拽自然流畅，并提供对拖拽后的位置记录等特性。而 jsMind 拖拽上略为生涩没有位置记录，不知后续是否有计划跟进？之所以如此冒昧请教，主要是基于项目选型上的考虑，jsMind 优势功能更完整，代码简洁易维护（轻量），GoJs - Mind Map 强在渲染，架构相对成熟易扩展。

**Comment by hizzgdev** [comment]
哈哈，首先感谢你对 jsMind 的支持与关注！
我之前并不了解 GoJs 这个项目，刚去简单试用了一下其中的 Mind Map，感觉其特点还是挺鲜明的，应该说有不同方向的侧重吧。

关于布局方面的考虑：jsMind 认为思维导图应该是有一个固定的布局模式的。比如，同样是从一个根节点出发不断扩展子节点的图形除了思维导图之外，还有目录结构、组织机构、鱼骨图等，这些图形内在的数据结构基本相同，展现方式却各有不同规则。所以未来可能不太会支持随意摆放结点这种能力。

关于拖拽方面的解释：在 jsMind 上对节点的拖拽其实是一种“修改”动作，作用是把一个节点移动到其它节点下，可以考虑一下如果支持随意摆放节点的话，如何修改节点的层级结构。

所以如果是直接回答你的这个问题的话：jsMind 后续是否有计划优化拖拽体验，并能保持拖拽后的位置？
那么目前没有相关计划，短期内应该也不会考虑提供这个能力。

如果是从项目选型的角度来说，你可以从项目本身的需求出发，看哪种功能是项目的本质需求。技术和产品没有好坏之分，关键在于合适，是吧。

**Reply by ipader** [reply]
感谢作者如此用心的回复。的确如此，功能定位和适用性更重要。开源的思维导图项目看似容易，实则五脏俱全，涉及前端技术庞杂，开发难度不小周期不短，不少项目因此陆续不再更新。作者能业余时间保持如此更新频率维持社区热度实属难得。

个人认为 jsMind 的定位清晰，基础模型及设计上已经有了不错的表现，若在用户体验、社区推动以及产品成熟度（第三方集成）上进一步发展，一定会很好的发展前景。会持续关注。再次感谢如此详复！

**Reply by hizzgdev** [reply]
感谢支持与鼓励！
